menas/src/main/scala/za/co/absa/enceladus/menas/HDFSConfig.scala

```diff
@@ -66,20 +62,31 @@ class HDFSConfig @Autowired()(spark: SparkSession) {
 
       logger.info("Using kerberos hadoop authentication")
 
-      System.setProperty("sun.security.krb5.debug", krb5debug);
-      System.setProperty("java.security.krb5.realm", krb5realm);
-      System.setProperty("java.security.krb5.kdc", krb5kdc);
+      System.setProperty("sun.security.krb5.debug", krb5debug)
+      System.setProperty("java.security.krb5.realm", krb5realm)
+      System.setProperty("java.security.krb5.kdc", krb5kdc)
 
 
       val c = hadoopConf()
-      
+
       SecurityUtil.setAuthenticationMethod(AuthenticationMethod.KERBEROS, c)
       UserGroupInformation.reset()
-      UserGroupInformation.setConfiguration(c);
+      UserGroupInformation.setConfiguration(c)
       UserGroupInformation.loginUserFromKeytab(krb5username, krb5keytab)
 
       FileSystem.get(c)
-      
-    } else FileSystem.get(hadoopConf)
+
+    } else if (authMethod == "simple") {
```

Comment:
This `if-else` situation is starting to look a lot like pattern-matching, you could also separate each of the cases into their own methods for readability.

Reply:
Thanks for the suggestion. Definitely can make the code more readable.